pso2 feature branch pull request: ready for testing

Shinico opened the pull request:
Please have a look at the command structure, otherwise ready for field-testing.

Changelog:
Added pso2 command
Imrpoved Help command functionality
Imrpoved Itemname functionality and added help text
Added shop command

Added static Utility Class

Namakamoto committed:
Improved pso2 (static methods für subcommands), gave functionality to itemname.js and added Utility

Improved pso2.js and added static methods reused by the subcommands.
Updated itemname.js with functionality.
Added Utility.js

Namakamoto committed:
Fixed itemname

Fixed itemname and added embed

Namakamoto committed:
pso2 itemname and shop added and improved

itemname and pso2 command help improved.
pso2 shop added

Marcedarc commented on commands/pso2.js:
```diff
@@ -16,6 +16,30 @@ export class pso2 extends ParentCommand {
-    "use `pso2 [item|itemname|shop] (<arguments>)` in order to receive help for the corresponding command.";
+    help = function(args, message) {
+        if(args[2] !== null && args[2] !== undefined && this.subCommands[args[2]] !== undefined && this.subCommands[args[2]] !== null) {
+            return this.subCommands.help;
```
you sure about this part?

Namakamoto committed:
Fixed imports and subCommand args

Fixed unnecessary imports
Fixed subCommand args

Marcedarc commented on custom_command_logic/pso2/itemname.js:
```diff
@@ -1,5 +1,8 @@
 import {BaseCommand} from "../util/BaseCommand";
 import {Message} from "discord.js";
+import  {DMChannel} from "discord.js";
```
Merge the 3 imported classes from one file into one like so:
`{Message, DMChannel, TextChannel}`
instead of using separate import statements

Shinico commented:
Fixed issues according to the review

Marcedarc commented on custom_command_logic/pso2/itemname.js:
```diff
@@ -14,12 +17,35 @@ export class itemname extends BaseCommand {
+        }
+        let replyFunction;
+        if(message.guild) {
+            replyFunction = function(m) {message.reply(m);};
```
Please avoid replying and mentioning users if not really necessary!
You can send the user's nickname if you need to address him as a direct answer to his own command.

Shinico replied:
Fixed in the latest commit.

Namakamoto committed:
Fixed itemname user mention

Fixed itemname user mention. Removed unnecessary if.

Marcedarc commented on commands/pso2.js:
```diff
@@ -16,6 +16,30 @@ export class pso2 extends ParentCommand {
      * @override
      * @type {string}
```
This looks more like
`@type {function}`
to me

Namakamoto committed:
Updated JS doc annotations

Updated JS doc annotations in order to comply to the new coding structure as well as the eye of the overseer.

Marcedarc commented on util/Utility.js:
```diff
@@ -0,0 +1,37 @@
+     *
+     * @param fromCh {number} Where the string chaining should start
+     * @param toCh {number} Where the string chaining should end
+     * @param args {array} String array to chain
```
please add a
`@return {type}`
statement

Namakamoto committed:
Fixed itemname user mention

Fixed itemname user mention. Removed unnecessary if.

Namakamoto committed:
Merge branch 'feature/pso2' of https://github.com/Komeran/avina into feature/pso2

Merge for the lulz

Namakamoto committed:
Fixed the annotations for real now.

Marcedarc reviewed (approved):
Nice and clean code
